test/integration/Client/client_lib_thread_interdependency_test.cpp

```diff
@@ -0,0 +1,499 @@
+    std::map <std::string, std::string> story_attrs;
+    flags = 2;
+    // Acquire story
+    auto acquire_ret = client->AcquireStory(chronicle_name, story_name, story_attrs, flags);
+    LOG_INFO("[ClientLibThreadInterdependencyTest] Story acquired: tid={}, Ret: {}", t->tid, acquire_ret.first);
+    handle_return_value(t->tid, acquire_ret.first, ThreadState::STORY_ACQUIRED);
+
+
+    // Release the story
+    ret = client->ReleaseStory(chronicle_name, story_name);
+    LOG_INFO("[ClientLibThreadInterdependencyTest] Story released: tid={}, Ret: {}", t->tid, ret);
+    handle_return_value(t->tid, ret, ThreadState::STORY_RELEASED);
+
+
+    // Destroy the story
+    ret = client->DestroyStory(chronicle_name, story_name);
+    LOG_INFO("[ClientLibThreadInterdependencyTest] Story destroyed: tid={}, Ret: {}", t->tid, ret);
+    handle_return_value(t->tid, ret, ThreadState::STORY_DESTROYED);
+
+
+    // Destroy the chronicle
+    ret = client->DestroyChronicle(chronicle_name);
+    LOG_INFO("[ClientLibThreadInterdependencyTest] Chronicle destroyed: tid={}, Ret: {}", t->tid, ret);
+    handle_return_value(t->tid, ret, ThreadState::CHRONICLE_DESTROYED);
+
+
+    // Thread Finalized
+    handle_return_value(t->tid, 0, ThreadState::THREAD_FINALIZED);
+}
+
+
+int main(int argc, char**argv)
+{
+    // Configuration
+    std::string conf_file_path = parse_conf_path_arg(argc, argv);
+    if(conf_file_path.empty())
+    {
+        std::exit(EXIT_FAILURE);
+    }
+    ChronoLog::ConfigurationManager confManager(conf_file_path);
+    int result = chronolog::chrono_monitor::initialize(confManager.CLIENT_CONF.CLIENT_LOG_CONF.LOGTYPE
+                                                       , confManager.CLIENT_CONF.CLIENT_LOG_CONF.LOGFILE
+                                                       , confManager.CLIENT_CONF.CLIENT_LOG_CONF.LOGLEVEL
+                                                       , confManager.CLIENT_CONF.CLIENT_LOG_CONF.LOGNAME
+                                                       , confManager.CLIENT_CONF.CLIENT_LOG_CONF.LOGFILESIZE
+                                                       , confManager.CLIENT_CONF.CLIENT_LOG_CONF.LOGFILENUM
+                                                       , confManager.CLIENT_CONF.CLIENT_LOG_CONF.FLUSHLEVEL);
+
+    if(result == 1)
+    {
+        std::exit(EXIT_FAILURE);
+    }
+
+    // Set up client & Connect
+    client = new chronolog::Client(confManager);
+    int ret = client->Connect();
+    if(chronolog::CL_SUCCESS != ret)
+    {
+        LOG_ERROR("[ClientLibThreadInterdependencyTest] Failed to connect to ChronoVisor");
+        delete client;
+        return -1;
+    }
+
+    // Initiate test
+    LOG_INFO("[ClientLibThreadInterdependencyTest] Running test.");
+    int num_threads = 8;
+    std::vector <struct thread_arg> t_args(num_threads);
+    std::vector <std::thread> workers(num_threads);
+    std::string client_id = gen_random(8);
```

Comment:
randomized client_id is not used and is not relevant for this test

Reply:
true